### PR TITLE
Add WavIntoSamples

### LINKED ISSUE #2: Add an equivalent to the WavSamples iterator that has ownership of the stream
If you take `Vec` for example, it as `iter()` which borrows the `Vec`, and `into_iter()` which takes ownership of it.

This library has `WavSamples`, which is an equivalent of `iter()`, but no equivalent to `into_iter()`.

### PR DESCRIPTION
**Pull request by tomaka**
Close #2 

I tackled it myself.

### CODE REVIEW

**Comment by tomaka** [comment]
Note that I have a work-around for this if you decide not to merge this PR.
This PR is mostly for convenience.

File: src/read.rs [hunk 1]
```diff
@@ -502,6 +511,16 @@ impl<R> WavReader<R> where R: io::Read {
         }
     }
 
+    /// Same as `samples`, but takes ownership of the `WavReader`.
+    ///
+    /// See `samples()` for more infos.
```
Comment (ruud-v-a) [inline src/read.rs]: Nit: infos → info (or information).

**Comment by ruud-v-a** [comment]
It makes sense to have this. Thank you for writing it, I’ll merge if you fix the typo.

**Comment by tomaka** [comment]
Done

**Comment by ruud-v-a** [comment]
Thanks!

**Pull request closed without merging**